### PR TITLE
wizard: Allow calendar date for restoration height

### LINKED ISSUE #1551: Allow calendar date for restoration height
Just as in Monerujo, very useful. Just like there, I propose the YYYY-MM-DD format.

### PR DESCRIPTION
**Pull request by mmbyday**
Fixes #1551
bh

### CODE REVIEW

File: wizard/WizardManageWalletUI.qml [hunk 1]
```diff
@@ -80,11 +80,16 @@ ColumnLayout {
...
             wizard.nextButton.enabled = checkFields();
         } else if (recoverMode && recoverFromSeedMode) {
-            wizard.nextButton.enabled = checkSeed()
+            wizard.nextButton.enabled = checkSeed() && checkRestoreHeight()
         } else
             wizard.nextButton.enabled = true;
     }
 
+    function checkRestoreHeight() {
+        console.log("Checking restoreHeight");
```
Comment (xiphon) [inline wizard/WizardManageWalletUI.qml]:
```suggestion
```

File: wizard/WizardManageWalletUI.qml [hunk 2]
```diff
@@ -297,18 +302,29 @@ ColumnLayout {
...
+        visible : recoverMode && recoverFromSeedMode
+        enabled : restoreHeight.indexOf('-') === 4 && restoreHeight.length === 10
+        onClicked: {
+            //parse date in YYYY-MM-DD format or return 0 as blockheight
+            restoreHeight = Utils.getApproximateBlockchainHeight(restoreHeight);
+            checkNextButton();
         }
     }
-    
     // Subaddress lookahead
```
Comment (xiphon) [inline wizard/WizardManageWalletUI.qml]:
```suggestion

    // Subaddress lookahead
```

File: wizard/WizardManageWalletUI.qml [hunk 3]
```diff
@@ -297,18 +302,29 @@ ColumnLayout {
...
             backgroundColor: "white"
             fontColor: "black"
             fontBold: false
+            validator: RegExpValidator {
+                regExp: /^(\d{1,4})*[-]?(\d{1,2})[-]?\d{1,2}$/
+            }
+            onTextUpdated: checkNextButton()
+        }
+    }
+    StandardButton {
```
Comment (xiphon) [inline wizard/WizardManageWalletUI.qml]:
```suggestion

    StandardButton {
```

File: wizard/WizardManageWalletUI.qml [hunk 4]
```diff
@@ -297,18 +302,29 @@ ColumnLayout {
...
             placeholderOpacity: 1.0
-            validator: IntValidator {
-                bottom:0
-            }
             borderColor: Qt.rgba(0, 0, 0, 0.15)
             backgroundColor: "white"
             fontColor: "black"
             fontBold: false
+            validator: RegExpValidator {
+                regExp: /^(\d{1,4})*[-]?(\d{1,2})[-]?\d{1,2}$/
```
Comment (xiphon) [inline wizard/WizardManageWalletUI.qml]:
```suggestion
                regExp: /^(\d+|\d{4}-\d{2}-\d{2})$/
```

**Comment by mmbyday** [comment]
All good, appreciate the feedback :)

File: wizard/WizardManageWalletUI.qml [hunk 5]
```diff
@@ -297,18 +301,31 @@ ColumnLayout {
             placeholderFontBold: true
             placeholderFontFamily: "Arial"
             placeholderColor: Style.legacy_placeholderFontColor
-            placeholderText: qsTr("Restore height (optional)") + translationManager.emptyString
+            placeholderText: qsTr("Wallet creation date in YYYY-MM-DD or Restore height (optional)") + translationManager.emptyString
```
Comment (xiphon) [inline wizard/WizardManageWalletUI.qml]: Worth to mention that a user might specify an approximate date

Comment (mmbyday) [reply wizard/WizardManageWalletUI.qml]: Any wording suggestions? Sounds like a good idea to me.

**Comment by selsta** [comment]
https://github.com/monero-project/monero-gui/blob/master/wizard/WizardCreateWalletFromDevice.qml#L63

Can you fix this too? If there’s no restore height set, 0 is used as the restore height. This should be changed to use `wallet2::estimate_blockchain_height()` (https://github.com/monero-project/monero/blob/master/src/wallet/wallet2.cpp#L3922), similar to how is in the CLI (https://github.com/monero-project/monero/blob/master/src/simplewallet/simplewallet.cpp#L3694).

**Comment by selsta** [comment]
Is there a reason to add a button for the conversion?

**Comment by mmbyday** [comment]
@selsta button no more! new PR #1892 :)

**Pull request closed without merging**